web/server/entrypoint.go

```diff
@@ -290,7 +302,13 @@ func (s *robotServer) createWebOptions(cfg *config.Config) (weboptions.Options,
 	if cfg.Cloud != nil && s.args.AllowInsecureCreds {
 		options.SignalingDialOpts = append(options.SignalingDialOpts, rpc.WithAllowInsecureWithCredentialsDowngrade())
 	}
-	options.SignalingDialOpts = append(options.SignalingDialOpts, rpc.WithSignalingConn(s.conn))
+
+	if s.signalingConn != nil {
```

Comment:
instead of doing this we could always set s.signalingConn, but set it to appConn unless `if cloud.SignalingAddress != "" && cloud.SignalingAddress != cloud.AppAddress`. That way logic is all kept in the block above and a bit more compartmentalized